### PR TITLE
[core] Fix function naming and error propagation

### PR DESCRIPTION
**Pull request by dentiny**
See inline comments for details.

### CODE REVIEW

File: src/ray/common/client_connection.h [hunk 1]
```diff
@@ -29,11 +29,13 @@
...
-typedef boost::asio::generic::stream_protocol local_stream_protocol;
-typedef boost::asio::basic_stream_socket<local_stream_protocol> local_stream_socket;
+using local_stream_protocol = boost::asio::generic::stream_protocol;
+using local_stream_socket = boost::asio::basic_stream_socket<local_stream_protocol>;
 
-void SetCloseOnFork(local_stream_socket &socket);
-void SetCloseOnFork(boost::asio::basic_socket_acceptor<local_stream_protocol> &acceptor);
+// Set "close on exec" feature for the given file descriptor.
+// WARNIGN: It does no-op on windows platform.
+void SetCloseOnExec(local_stream_socket &socket);
```
Comment (dentiny) [inline src/ray/common/client_connection.h:37]: Function naming is incorrect, `fork` != `exec`

File: src/ray/common/function_descriptor.h [hunk 2]
```diff
@@ -129,9 +130,11 @@ class JavaFunctionDescriptor : public FunctionDescriptorInterface {
   }
 
   virtual std::string ToString() const {
-    return "{type=JavaFunctionDescriptor, class_name=" + typed_message_->class_name() +
```
Comment (dentiny) [inline src/ray/common/function_descriptor.h:132]: bad code, it creates multiple intermediate temporary strings

File: src/ray/common/grpc_util.h [hunk 3]
```diff
@@ -38,23 +39,21 @@ class MessageWrapper {
   /// The input message will be **copied** into this object.
   ///
   /// \param message The protobuf message.
-  explicit MessageWrapper(const Message &message)
```
Comment (dentiny) [inline src/ray/common/grpc_util.h:41]: Checkout modern effective C++ item41, passing by value is the best way to 
- handle both lvalue and rvalue overload
- with ownership held

File: src/ray/common/grpc_util.h [hunk 4]
```diff
@@ -99,19 +97,20 @@ inline std::string GrpcStatusToRayStatusMessage(const grpc::Status &grpc_status)
 inline Status GrpcStatusToRayStatus(const grpc::Status &grpc_status) {
   if (grpc_status.ok()) {
     return Status::OK();
-  } else if (grpc_status.error_code() == grpc::StatusCode::DEADLINE_EXCEEDED) {
```
Comment (dentiny) [inline src/ray/common/grpc_util.h:102]: Follows https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html

**Review by jjyao — APPROVED** [review]

File: src/ray/common/client_connection.cc [hunk 5]
```diff
@@ -59,25 +59,26 @@ void setFdCloseOnFork(const NativeHandleType &handle) {
...
 // See https://github.com/ray-project/ray/issues/40813
-void setFdCloseOnFork(int fd) {
+void SetFdCloseOnExec(int fd) {
   if (fd < 0) {
     return;
   }
   int flags = fcntl(fd, F_GETFD, 0);
-  RAY_CHECK(flags != -1) << "fcntl error: errno = " << errno << ", fd = " << fd;
-  fcntl(fd, F_SETFD, flags | FD_CLOEXEC);
+  RAY_CHECK_NE(flags, -1) << "fcntl error: errno = " << errno << ", fd = " << fd;
```
Comment (dentiny) [inline src/ray/common/client_connection.cc:67]: We should have macros to check errnos.

**Pull request merged**